### PR TITLE
Add information how to see the features of the dataset

### PR DESCRIPTION
**Pull request by adam-narozniak**
## Issue
The feature information (the name of the image label - either "img" or "image" that differs on HF) is needed to make decisions on the transformation of the dataset. Yet, it's not emphasized enough in the docs.

## Proposal
Improve the documentation: quickstart + all the how-to-{framework} sections to include this information. 

This PR also slightly restructures the TensorFlow example by pulling the common part to the beginning and not repeating it (there not repeating the newly added section neither)

### Explanation
It is needed:
* to apply transforms in PyTorch, 
* divided the dataset into `X` and `y` in NumPy,
* transform into `TensorFlow Dataset` in TensorFlow

### CODE REVIEW

File: datasets/doc/source/how-to-use-with-numpy.rst [hunk 1]
```diff
@@ -3,14 +3,30 @@ Use with NumPy
...
-Prepare the desired partitioning::
+Create a ``FederatedDataset``::
 
   from flwr_datasets import FederatedDataset
 
   fds = FederatedDataset(dataset="cifar10", partitioners={"train": 10})
   partition = fds.load_partition(0, "train")
   centralized_dataset = fds.load_full("test")
 
+Determine the names of the features::
```
Comment (jafermarq) [inline datasets/doc/source/how-to-use-with-numpy.rst:14]:
```suggestion
Inspect the names of the features::
```

File: datasets/doc/source/how-to-use-with-tensorflow.rst [hunk 2]
```diff
@@ -1,10 +1,32 @@
...
 
+Create a ``FederatedDataset``::
+
+  from flwr_datasets import FederatedDataset
+
+  fds = FederatedDataset(dataset="cifar10", partitioners={"train": 10})
+  partition = fds.load_partition(0, "train")
+  centralized_dataset = fds.load_full("test")
+
+Determine the names of the features::
```
Comment (jafermarq) [inline datasets/doc/source/how-to-use-with-tensorflow.rst:16]:
```suggestion
Inspect the names of the features::
```

File: datasets/doc/source/how-to-use-with-numpy.rst [hunk 3]
```diff
@@ -3,14 +3,30 @@ Use with NumPy
...
+
+In case of CIFAR10, you should see the following output.
+
+.. code-block:: none
+
+  {'img': Image(decode=True, id=None),
+  'label': ClassLabel(names=['airplane', 'automobile', 'bird', 'cat', 'deer', 'dog',
+  'frog', 'horse', 'ship', 'truck'], id=None)}
+
+Let's move to the transformations.
```
Comment (jafermarq) [inline datasets/doc/source/how-to-use-with-numpy.rst:26]:
```suggestion
We will use the keys in the partition features in order to apply transformations to the data or pass it to a ML model.  Let's move to the transformations.
```

File: datasets/doc/source/how-to-use-with-tensorflow.rst [hunk 4]
```diff
@@ -1,10 +1,32 @@
...
+
+In case of CIFAR10, you should see the following output.
+
+.. code-block:: none
+
+  {'img': Image(decode=True, id=None),
+  'label': ClassLabel(names=['airplane', 'automobile', 'bird', 'cat', 'deer', 'dog',
+  'frog', 'horse', 'ship', 'truck'], id=None)}
+
+Let's move to the transformations.
```
Comment (jafermarq) [inline datasets/doc/source/how-to-use-with-tensorflow.rst:28]:
```suggestion
We will use the keys in the partition features in order to construct a `tf.data.Dataset <https://www.tensorflow.org/api_docs/python/tf/data/Dataset>_`. Let's move to the transformations.
```

Comment (jafermarq) [reply datasets/doc/source/how-to-use-with-tensorflow.rst:28]: I left the "Let's move to the transformations." sentence but the doc page never shows how to apply transforms

File: datasets/doc/source/how-to-use-with-tensorflow.rst [hunk 5]
```diff
@@ -33,17 +45,7 @@ Transformation to the TensorFlow Dataset::
...
-Standard setup::
-
-  from flwr_datasets import FederatedDataset
-
-  fds = FederatedDataset(dataset="cifar10", partitioners={"train": 10})
-  partition = fds.load_partition(0, "train")
-  centralized_dataset = fds.load_full("test")
-
-Transformation to the TensorFlow Tensors ::
+Transform the data to the ``TensorFlow Tensors`` (it's not the TensorFlow dataset)::
```
Comment (jafermarq) [inline datasets/doc/source/how-to-use-with-tensorflow.rst:48]:
```suggestion
Transform the data to the TensorFlow `tf.Tensor <https://www.tensorflow.org/api_docs/python/tf/Tensor>`_ (it's not the TensorFlow dataset)::
```

Comment (jafermarq) [reply datasets/doc/source/how-to-use-with-tensorflow.rst:48]: better to point to docs?

File: datasets/doc/source/tutorial-quickstart.rst [hunk 6]
```diff
@@ -5,7 +5,7 @@ Run Flower Datasets as fast as possible by learning only the essentials.
 
 Install Federated Datasets
 --------------------------
-Run on the command line
+On the command line, run
 
 .. code-block:: bash
```
Comment (jafermarq) [inline datasets/doc/source/tutorial-quickstart.rst:10]: Should we indicate that:
```
python -m pip install flwr-datasets[vision]
```
might not work depending on the shell you use? for instnace for me on `zsh` using `[` never works. I need to scape them:
```
python -m pip install flwr-datasets\[vision\]
```

Maybe worth having a comment like: "depending on your shell, you might need to wrap `[vision]` with `\[vision\]`

**Review by jafermarq — COMMENTED** [review]
I left some comments! 

Although not part of this PR, i think the `Use with PyTorch` page would benefit from some sections/subsections just like you do for TF and NumPy

File: datasets/doc/source/how-to-use-with-tensorflow.rst (hunk 4 above)
Comment (adam-narozniak) [reply datasets/doc/source/how-to-use-with-tensorflow.rst:28]: I mean the transformation of the format from Dataset to NumPy (not in the sense of PyTorch transformations). But I'm fine with changing it or expressing it more clearly

File: datasets/doc/source/tutorial-quickstart.rst (hunk 6 above)
Comment (adam-narozniak) [reply datasets/doc/source/tutorial-quickstart.rst:10]: Does it work if you make it a string? so `"flwr-datasets[vision]"`

File: datasets/doc/source/how-to-use-with-tensorflow.rst (hunk 5 above)
Comment (adam-narozniak) [reply datasets/doc/source/how-to-use-with-tensorflow.rst:48]: It's a good reference

File: datasets/doc/source/tutorial-quickstart.rst (hunk 6 above)
Comment (jafermarq) [reply datasets/doc/source/tutorial-quickstart.rst:10]: yes, it works

File: datasets/doc/source/how-to-use-with-tensorflow.rst (hunk 4 above)
Comment (jafermarq) [reply datasets/doc/source/how-to-use-with-tensorflow.rst]: oh, got it

File: datasets/doc/source/tutorial-quickstart.rst (hunk 6 above)
Comment (adam-narozniak) [reply datasets/doc/source/tutorial-quickstart.rst:10]: So, let me change it to a string.

**Review by danieljanes — APPROVED** [review]
